bundles/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/weather/NetatmoWeatherBinding.java

```diff
@@ -0,0 +1,557 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.netatmo.internal.weather;
+
+import static org.openhab.binding.netatmo.internal.weather.MeasurementRequest.createKey;
+
+import java.math.BigDecimal;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.openhab.binding.netatmo.NetatmoBindingProvider;
+import org.openhab.binding.netatmo.internal.NetatmoBinding;
+import org.openhab.binding.netatmo.internal.NetatmoException;
+import org.openhab.binding.netatmo.internal.authentication.OAuthCredentials;
+import org.openhab.binding.netatmo.internal.messages.NetatmoError;
+import org.openhab.binding.netatmo.internal.weather.GetStationsDataResponse.Device;
+import org.openhab.binding.netatmo.internal.weather.GetStationsDataResponse.Module;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Binding that gets measurements from the Netatmo API every couple of minutes.
+ *
+ * @author Andreas Brenk
+ * @author Thomas.Eichstaedt-Engelen
+ * @author Gaël L'hopital
+ * @author Rob Nielsen
+ * @author Ing. Peter Weiss
+ * @since 1.4.0
+ */
+public class NetatmoWeatherBinding {
+
+    private static final String WIND = "Wind";
+
+    private static final Logger logger = LoggerFactory.getLogger(NetatmoWeatherBinding.class);
+
+    protected static final String CONFIG_PRESSURE_UNIT = "pressureunit";
+    protected static final String CONFIG_UNIT_SYSTEM = "unitsystem";
+
+    private Map<Device, PointType> stationPositions = new HashMap<Device, PointType>();
+
+    private NetatmoPressureUnit pressureUnit = NetatmoPressureUnit.DEFAULT_PRESSURE_UNIT;
+    private NetatmoUnitSystem unitSystem = NetatmoUnitSystem.DEFAULT_UNIT_SYSTEM;
+
+    private GetStationsDataRequest stationsDataRequest;
+    private GetStationsDataResponse stationsDataResponse;
+
+    /**
+     * Execute the weather binding from Netatmo Binding Class
+     *
+     * @param oauthCredentials
+     * @param providers
+     * @param eventPublisher
+     */
+    public void execute(OAuthCredentials oauthCredentials, Collection<NetatmoBindingProvider> providers,
+            EventPublisher eventPublisher) {
+        logger.debug("Querying Netatmo Weather API");
+
+        try {
+            if (oauthCredentials.isFirstExecution()) {
```

Comment:
oauthCredentials is used by both weather and camera, this check needs to use a private global variable in this binding to indicate that is the first time used.